conference.js

```diff
@@ -1664,27 +1664,47 @@ export default {
             return;
         }
 
+        // Create a new presenter track and apply the presenter effect.
         if (!this.localPresenterVideo && !mute) {
-            // create a new presenter track and apply the presenter effect.
             let { height } = this.localVideo.track.getSettings();
-
-            // Workaround for Firefox since it doesn't return the correct width/height of the desktop stream
-            // that is being currently shared.
-            if (!height) {
-                const desktopResizeConstraints = {
+            const { aspectRatio, width } = this.localVideo.track.getSettings();
+            let desktopResizeConstraints = {};
+            let resizeDesktopStream = false;
+
+            // Determine the constraints if the desktop track needs to be resized.
+            // Resizing is needed when the resolution cannot be determined or when
+            // the window is bigger than 1280x720.
+            if (height && width && aspectRatio) {
+                const advancedConstraints = [ { aspectRatio } ];
+                const resizeHeight = height >= width;
+
+                // Determine which dimension needs resizing and resize only that side
```

Comment:
The logic here doesn't look correct. If you have a portrait 720p screen it means it's 720x1280.  Now imagine it's a tad bigger, say 800x1280. The code below will resize the height tto 720, but that's not what you want, is it?  What we want is to resize the width to be 720.

```
// Somewhere else:
const DESKTOP_STREAM_CAP = 720;

const isPortrait = height >= width;
if (isPortrait && width > DESKTOP_STREAM_CAP) {
    resizeDesktopStream = true;
    advancedConstraints.push({ width: DESKTOP_STREAM_CAP });
} else if (!isPortrait && height > DESKTOP_STREAM_CAP) {
    resizeDesktopStream = true;
    advancedConstraints.push({ height: DESKTOP_STREAM_CAP });
}
```

Isn't the above what we're after?

Reply:
Since we are using height for determining the resolution of the camera, I was making sure it is not greater than 720 but I see your point.